boards: arduino_due: Add arduino_i2c alias

zerkerX opened the pull request:
The arduino_i2c alias is needed to use the Arduino Due with some of the
examples. This is assigned to the twi1 interface, which is labelled as
SDA and SCL (20 and 21) on the board silkscreen and is the default
interface using the Arduino IDE.

Resolves #42881.

Signed-off-by: Ryan Armstrong <git@zerker.ca>

zerkerX committed:
boards: arduino_due: Add arduino_i2c alias

The arduino_i2c alias is needed to use the Arduino Due with some of the
examples. This is assigned to the twi1 interface, which is labelled as
SDA and SCL (20 and 21) on the board silkscreen and is the default
interface using the Arduino IDE.

Resolves #42881.

Signed-off-by: Ryan Armstrong <git@zerker.ca>

nandojve commented on boards/arm/arduino_due/arduino_due.dts:
```diff
@@ -56,3 +56,5 @@
 	bias-pull-up;
 };
+
+arduino_i2c: &twi1 { };
```
LGTM! Maybe it is important make sure twister will test this config by adding a filter like this: https://github.com/zephyrproject-rtos/zephyr/blob/53ce87a9721d5172146f1f6d41eec55bf910ff4d/boards/arm/sam_v71_xult/sam_v71_xult.yaml#L13

zerkerX committed:
boards: arduino_due: Add arduino_i2c to support list

Correcting pull request review finding to add `arduino_i2c` to the list
of supported items in the .yaml file.

Signed-off-by: Ryan Armstrong <git@zerker.ca>

nandojve reviewed (changes requested):
See change request below:

nandojve commented on drivers/i2c/i2c_sam_twi.c:
```diff
@@ -66,6 +66,8 @@ struct i2c_sam_twi_dev_data {
 	struct twi_msg msg;
 };
 
+#define DEV_NAME(dev) ((dev)->name)
```
NAK on this!

Please, open an issue about it and send a PR that drop remaining DEV_.

zerkerX replied:
Alright. I was probably jumping a bit ahead to address the integration issue. I've raised issue #42996. Let me just revert this commit from the pull request...

nandojve commented:
Hi @zerkerX ,

Can you rebase your commits on top of main and force a push.

nandojve reviewed (approved):
LGTM!
